server/Controllers/UserController.cs

```diff
@@ -0,0 +1,118 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.AspNetCore.Identity;
+using server.Models;
+
+namespace server.Controllers;
+
+[ApiController]
+[Route("[controller]")]
+public class UserController : ControllerBase
+{
+    private UserManager<User> userManager;
+
+    public UserController(ServerContext context, UserManager<User> userMgr)
+    {
+        userManager = userMgr;
+    }
+
+    [HttpGet]
+    public async Task<ActionResult<IEnumerable<GetUser>>> GetUsers()
+    {
+        return HandleResults(await userManager.Users.ToListAsync());
+    }
+
+    [HttpPost]
+    public async Task<ActionResult<GetUser>> CreateUser(PostUser data)
+    {
+        if (ModelState.IsValid)
+        {
+            User user = new User
+            {
+                UserName = data.UserName,
+                Email = data.Email,
+                FullName = data.FullName
+            };
+
+            return HandleResults(user, await userManager.CreateAsync(user, data.Password));
+        }
+
+        return UnprocessableEntity(ModelState);
+    }
+
+    [HttpGet("{email}")]
+    public async Task<ActionResult<GetUser>> GetUser(string email)
+    {
+        var user = await userManager.FindByEmailAsync(email);
+
+        if (user == null)
+            return NotFound();
+
+        return HandleResults(user);
+    }
+
+    [HttpPut("{email}")]
+    public async Task<ActionResult<GetUser>> UpdateUser(string email, PutUser data)
+    {
+        var user = await userManager.FindByEmailAsync(email);
+
+        if (user == null)
+            return NotFound();
+
+        user.UserName = String.IsNullOrEmpty(data.UserName) ? user.UserName : data.UserName;
+        user.Email = String.IsNullOrEmpty(data.Email) ? user.Email : data.Email;
+        user.FullName = String.IsNullOrEmpty(data.FullName) ? user.FullName : data.FullName;
+

```

Comment:
suggestion: we can use auto mapper for this https://github.com/AutoMapper/AutoMapper